tests/unit/test_jlink.py

```diff
@@ -935,9 +946,10 @@ def test_jlink_close_context_manager(self):
           ``None``
         """
         # Use open_tunnel=None to avoid implicitly opening the connection.
-        with jlink.JLink(self.lib, open_tunnel=None):
-            self.dll.JLINKARM_OpenEx.assert_not_called()
+        with jlink.JLink(self.lib, open_tunnel=None) as jl:
             self.dll.JLINKARM_Close.assert_not_called()
+            # Bump the refcount without calling .open().
+            jl._open_refcount = 1
```

Comment:
Why do we bump the ref count here?

Reply:
The intent of this test is to verify that .close() gets called when the context manager exits. The refcount gets increased on calls to .open(), but the way this test is set up .open() is never called (see the comment on line 948, which I believe results in no .open() call).

I bump the refcount here manually instead. If there's a straightforward way to open a JLink object in a context manager and have it call .open() when doing so, I'd be happy to use that approach.

Reply:
Ah, that makes sense.  Sorry about that!